apt/minimization/dtimportanceminimizer.py

```diff
@@ -0,0 +1,24 @@
+from . import OrderedFeatureMinimizer
+from sklearn.tree import DecisionTreeClassifier
+import numpy as np
+
+
+class DTImportanceMinimizer(OrderedFeatureMinimizer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, ordered_features=None, **kwargs)
+        self._importance_ordered_features = None
+
+    @property
+    def importance_ordered_features(self):
+        return self._importance_ordered_features
+
+    def _get_ordered_features(self, estimator, encoder, X_train, y_train, numerical_features, categorical_features,
```

Comment:
Looks like the order is only computed when calling an internal method (begins with _). Please change this to be an external method (with documentation), and if it receives x and y data better to call it fit.

Reply:
This method is a hook that is called in the fit method in the superclass (OrderedFeatureMinimizer). It is a protected method that should be overriden to define a custom order to prune according during fit.

The implementation as it is allows access to the order through a property only after fit() is called, the property returns None otherwise. This property has a different name for different classes (here it is importace_ordered_features). I could add this to OrderedFeatureMinimizer as a generic property called features_order if you believe that is more suitable.

Reply:
Ok, I understand. I suggest to have a consistent naming for the property. Also, at the moment the fit code does not set self._ordered_features, which I think it should (instead of setting a local variable).